mkosi/__init__.py

```diff
@@ -2241,6 +2242,100 @@ def install_openmandriva(args: CommandLineArguments, root: str, do_run_build_scr
     disable_pam_securetty(root)
 
 
+@complete_step("Installing Gentoo")
+def install_gentoo(args: CommandLineArguments, root: str, do_run_build_script: bool) -> None:
+    try:
+        import multiprocessing
+
+        from _emerge.actions import (
```

Comment:
interesting, it would be the first time we do this, i.e. use the packaging libs directly. But I must say, I kinda sympathize with this. And we probably should do the same for other OSes where we can, i.e. use the dnf python modules on fedora for example.

Why that? I think in the dnf case at least this opens up new possibilities for us, i.e. we could add support for "pinning" package version: i.e. think that you first build your OS like you always do, but we output RPM package version data to some file while doing so. Next time when we rebuild we read that file and make sure to pick the precise same versions of the RPMs. And if you drop the file you basically update to current RPMs. Something like that would hopefully bring us a bit closer to reproducible builds in the long run.

Reply:
Sounds good, we can just depend on these being installed alongside the corresponding package manager.

@257 are the Python files part of the default emerge install?